src/SOS/SOS.Hosting/SOSHost.cs

```diff
@@ -1057,21 +1057,21 @@ private bool SearchVersionString(ModuleInfo moduleInfo, out string fileVersion)
             IntPtr self,
             ulong* offset)
         {
-            if (!AnalyzeContext.CurrentThreadId.HasValue)
-            {
-                return E_FAIL;
-            }
-            uint threadId = AnalyzeContext.CurrentThreadId.Value;
-            try
+            if (AnalyzeContext.CurrentThreadId.HasValue)
             {
-                ulong teb = _threadService.GetThreadInfoFromId(threadId).ThreadTeb;
-                Write(offset, teb);
-                return S_OK;
-            }
-            catch (InvalidOperationException)
-            {
-                return E_FAIL;
+                uint threadId = AnalyzeContext.CurrentThreadId.Value;
+                try
+                {
+                    ulong teb = _threadService.GetThreadInfoFromId(threadId).ThreadTeb;
+                    Write(offset, teb);
+                    return S_OK;
+                }
+                catch (DiagnosticsException)
+                {
+                }
             }
+            Write(offset, 0);
```

Comment:
Is it common for a TEB to be missing like this? Otherwise, should we log an error on this path?

Reply:
The exception is thrown because the thread id is invalid not anything to do with the TEB. So it shouldn't be that common.